### DISCUSSION TITLE
IOBroker possible over MQTT?

### DISCUSSION THREAD

**Discussion by openKimchi** [revk/xESP32-Faikout#162] [General]
I do not use Home Assistant but IOBroker. Is there a chance to get values over MQTT also?

**Comment by revk** [comment]
The MQTT reporting and control provides values in more than one way, the HA stuff specifically provides config for HA. But if what you have can be told to look for values in MQTT messages, you should be able to. Look at the various messages posted to MQTT, and in the manual. They work using JSON which most things understand. I have no specific knowledge of IOBroker, others may be able to help.

**Comment by openKimchi** [comment]
OK thanks!
I have now connected Faikin to an MQTT server (Mosquito). I see the topics in the MQTT Explorer. Now it should be possible to create a topic here that I can use to turn off the air conditioning, for example. I tried with the topic command/MyFaikin/command with the value off. But that doesn't work.

**Comment by revk** [comment]
Assuming `MyFaikin` is the name you gave it, `command/MyFaiikin/off` should work.

**Reply by openKimchi** [reply]
OK, it is not MyFaikin but FaikinWohnzimmer. MyFaikin was only a placeholder.
Here is a what I published:
![image](https://github.com/revk/ESP32-Faikin/assets/118620843/89456b1a-e3ae-4b92-981a-f70af2b7f105)
That does unfortunately not work.

**Comment by revk** [comment]
Your are seeing a topic of `command/FaikinWohnzimmer` and a payload of `off`.

Try sending a topic of `comamnd/FaikinWohnzimmer/off` (no payload).

Some commands do have a payload, such as `command/FaikinWohnzimmer/temp` where the payload is the desired temperature. But the `off` command has no payload.

**Reply by openKimchi** [reply]
OK, I sent a topic:
![image](https://github.com/revk/ESP32-Faikin/assets/118620843/057f6c6a-f7e4-4566-b0d0-7c73afb03e45)

nothing happens...

in ioBroker you can see the MQTT States:
![image](https://github.com/revk/ESP32-Faikin/assets/118620843/7da89ea4-76c9-410e-8e5a-76b91b2bcaec)

if I now delete the state command/FaikinWohnzimmer/off the AC turns off....very strange...

**Reply by revk** [reply]
It is tricky to know exactly what you are sending and when to be honest, that `off` command should turn the aircon off. How that might interact with ioBroker is not something I know.

The `off` command is not a "state" really, it is a command, a "make this change now" things, so that may be some confusion.

If you need to pass some sort of state / variable you may want to do `command/FaikinWohnzimmer/control` with a payload of the form `{"power":0}` for off or `{"power":1}` for on, which may be something you can program as a `{"power":$variable}` or something, but I really don't know ioBroker or how it works, so that is just a guess.

**Reply by openKimchi** [reply]
OK thank you!
I just discovered that I can control the target temperature using command/FaikinWohnzimmer/temp and the payload. Also from ioBroker with a script.
That's more than enough for me, I don't really need on/off.

**Reply by revk** [reply]
Excellent, have fun.
